commands/eval.b.inc

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @file
+ * Command(s) for evaluating PHP code directly.
+ */
+
+/**
+ * Implements hook_b_command().
+ */
+function eval_b_command() {
+  return array(
+    'php-eval' => array(
+      'description' => bt('Evaluate arbitrary PHP code after bootstrapping Backdrop.'),
+      'callback' => 'php_eval_b_command',
+      'arguments' => array(
+        'code' => bt('The PHP code to evaluate.'),
+      ),
+      'aliases' => array('ev', 'eval'),
+      'bootstrap' => B_BOOTSTRAP_FULL,
+      'examples' => array(
+        'b php-eval \'$node = node_load(1); print $node->title;\'' => 'Loads node with nid 1 and then prints its title.',
```

Comment:
Is there a reason to surround the PHP code with single quotes and then escape them? Seems it'd just be easier to use double-quotes instead... Also, example descriptions should be translatable.

```suggestion
        'b php-eval "$node = node_load(1); print $node->title;"' => bt('Loads node with nid 1 and then prints its title.'),
```

Reply:
Yes, there's a very good reason for the single quotes (and it's bitten me before). When you're doing a CLI command, the shell gets first crack at parsing before it passes the arguments to b, and if the shell sees a $ sign inside of a double-quoted string, it's going to try to substitute a value for that variable (just like PHP does), with likely unexpected results. If you use a double-quoted string, then b will never see "$node", it will see whatever the shell tries to substitute for $node (usually an empty string). 

So, if you try that command with the double-quoted version, it will fail (and in a very puzzling way).

Reply:
Ok. So I had thought that might be the case, but when I looked at the third example, it has `$HOME` in double quotes, so I figured that'd be ok for the first example too. But I suppose `$HOME` is a bash variable, whereas `$node` is a PHP variable, and so they're handled differently...

Reply:
They're actually handled the same way; anything with a $ could be a bash variable and is treated as such inside a double-quoted string. So, for example, if I had in my bash script
```
node=foobar
b ev "print $node->title;"
```
then b would actually attempt to evaluate `'print foobar->title;'`

I could even do
```
node=\$user
b ev "print $node->name'"
```
and then b would actually evaluate `'print $user->name;'`

The moral being, I guess, stick with single quotes unless you're specifically after variable substitution.

But yes, $HOME is a predefined bash variable, so it will always get substituted.